ghcide/src/Development/IDE/Core/Compile.hs

```diff
@@ -474,13 +473,13 @@ generateHieAsts hscEnv tcm =
 
 spliceExpresions :: Splices -> [LHsExpr GhcTc]
 spliceExpresions Splices{..} =
-    DL.toList $ mconcat
-        [ DL.fromList $ map fst exprSplices
-        , DL.fromList $ map fst patSplices
-        , DL.fromList $ map fst typeSplices
-        , DL.fromList $ map fst declSplices
-        , DL.fromList $ map fst awSplices
-        ]
+  fold
+    [ map fst exprSplices
+    , map fst patSplices
+    , map fst typeSplices
+    , map fst declSplices
+    , map fst awSplices
+    ]
```

Comment:
Thinking that marshaling the: `[[xs]] -> n fromList -> DList.concat -> toList -> [xs]` - is controversial. I noted superfluous back & forth conversion & looked into official docs.

The commit message point to the official docs which mention that conversion to/fromList are also `n` constly, in https://hackage.haskell.org/package/dlist-1.0/docs/Data-DList.html

> `fromList`: This function is implemented with '++'. Repeated uses of @fromList@ are just as inefficient as repeated uses of '++'.

Because DList Haddocks do not have tags to point into `to/fromList` - in the commit message I could only point to the library source.
 
So the code was: every `fromList` is `++` (means `<>` of `[]` ) already & then with additionally added `DList.concat` on top for them. So, more optimal is just fold with `++`, which is `fold` on `[]`.